mleap-core/src/main/scala/ml/combust/mleap/core/types/Casting.scala

```diff
@@ -184,13 +184,41 @@ object Casting {
                   val cc = c.asInstanceOf[(Any) => ByteString]
                   (v: Any) => Tensor.denseVector(v.asInstanceOf[Seq[_]].map(cc).toArray)
               }
+            }
+          }
+        }.orElse {
+          Some {
+            Try {
+              from.base match {
+                case BasicType.Boolean =>
+                  (v: Any) => Tensor.denseVector(v.asInstanceOf[Seq[_]].map(_.asInstanceOf[Boolean]).toArray)
+                case BasicType.Byte =>
+                  (v: Any) => Tensor.denseVector(v.asInstanceOf[Seq[_]].map(_.asInstanceOf[Byte]).toArray)
+                case BasicType.Short =>
+                  (v: Any) => Tensor.denseVector(v.asInstanceOf[Seq[_]].map(_.asInstanceOf[Short]).toArray)
+                case BasicType.Int =>
+                  (v: Any) => Tensor.denseVector(v.asInstanceOf[Seq[_]].map(_.asInstanceOf[Int]).toArray)
+                case BasicType.Long =>
+                  (v: Any) => Tensor.denseVector(v.asInstanceOf[Seq[_]].map(_.asInstanceOf[Long]).toArray)
+                case BasicType.Float =>
+                  (v: Any) => Tensor.denseVector(v.asInstanceOf[Seq[_]].map(_.asInstanceOf[Float]).toArray)
+                case BasicType.Double =>
+                  (v: Any) => Tensor.denseVector(v.asInstanceOf[Seq[_]].map(_.asInstanceOf[Double]).toArray)
+                case BasicType.String =>
+                  (v: Any) => Tensor.denseVector(v.asInstanceOf[Seq[_]].map(_.asInstanceOf[String]).toArray)
+                case BasicType.ByteString =>
+                  (v: Any) => Tensor.denseVector(v.asInstanceOf[Seq[_]].map(_.asInstanceOf[ByteString]).toArray)
+              }
+            }
           }
         }
       case (_: TensorType, _: ListType) =>
         baseCast(from.base, to.base).map {
-          _.map {
-            c => (v: Any) => v.asInstanceOf[Tensor[_]].rawValues.toList.map(c)
+          _.flatMap {
+            c => Try((v: Any) => v.asInstanceOf[Tensor[_]].rawValues.toList.map(c))
           }
+        }.orElse {
+          Some(Try((v: Any) => v.asInstanceOf[Tensor[_]].rawValues.toList))
```

Comment:
This works without manually specifying `_.asInstanceOf` like we did in the (List -> Tensor) direction

I believe this is because `Tensor` is not a native Scala class, so casting to one doesn't automatically cast all of the elements into Java `Object` types